ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Query.hs

```diff
@@ -44,23 +51,32 @@ import           Ouroboros.Consensus.Util.DepPair
 
 -- | Different queries supported by the ledger for all block types, indexed
 -- by the result type.
-data Query blk result
-  = BlockQuery (BlockQuery blk result)
-  -- ^ This constructor is supported by all @QueryVersion@s. The @BlockQuery@
+data Query blk result where
+  -- | This constructor is supported by all @QueryVersion@s. The @BlockQuery@
   -- argument is versioned by the @BlockNodeToClientVersion blk@.
+  BlockQuery :: BlockQuery blk result -> Query blk result
+
+  -- | Get the 'SystemStart' time.
```

Comment:
Add to the comment saying something like `Supported by query version >= QueryV1`